Automattic/wp-calypso
# Tags page: Add top section of public tags page

Add a `/tags` page to replace the current page at wordpress.com/tags

see project thread: pe7F0s-Ed-p2

Only includes the page header and the first part of the page. I will add the alphabetic tags section in a follow up PR:

<img width="1263" alt="Screen Shot 2023-03-28 at 2 33 58 pm" src="https://user-images.githubusercontent.com/22446385/228146281-f6904020-acdd-4a7c-8c57-07e8141f5d7f.png">

This page is made available to search engines via SSR. see SSR docs 
https://github.com/Automattic/wp-calypso/blob/trunk/docs/isomorphic-routing.md
https://github.com/Automattic/wp-calypso/blob/trunk/docs/server-side-rendering.md

### Testing instructions
* The special page header should only be applied to the new `/tags` page. It should not affect headers on other pages such as `/read` (normal logged in masterbar) or `/themes` (when logged out).
* There is a logged out version and a logged in version of the page, with different headers. p1679967423624639-slack-C03NLNTPZ2T
* It is possible to see the server side rendered version of the page by doing _View Page Source_ and searching the generated HMTL
